### DISCUSSION TITLE
Use EIP-1191 next breaking change

### DISCUSSION THREAD

**Discussion by roninjin10** [wevm/wagmi#3912] [Ideas]
Wagmi does not pass the chainId into `getAddress`. Example: https://github.com/wevm/wagmi/blob/8fc039d66fe59c45b5b5a6a36f691ea8174a82b6/packages/connectors/src/metaMask.ts#L113

Passing the chainId will overall be a better ux default but since it's a breaking change this change should happen on a breaking release

## Links

EIP-1191: https://github.com/ethereum/ercs/blob/master/ERCS/erc-1191.md
getAddress: https://viem.sh/docs/utilities/getAddress#chainid-optional

**Comment by sambacha** [comment]
> Wagmi does not pass the chainId into `getAddress`. Example: https://github.com/wevm/wagmi/blob/8fc039d66fe59c45b5b5a6a36f691ea8174a82b6/packages/connectors/src/metaMask.ts#L113
> 
> 
> 
> Passing the chainId will overall be a better ux default but since it's a breaking change this change should happen on a breaking release
> 
> 
> 
> ## Links
> 
> 
> 
> EIP-1191: https://github.com/ethereum/ercs/blob/master/ERCS/erc-1191.md
> 
> getAddress: https://viem.sh/docs/utilities/getAddress#chainid-optional


Can we do replay protection extension at the contract level and provide a stronger guarantee potentially? 


~~~solidity

// constructor 
deploymentChainId = block.chainid;
_DOMAIN_SEPARATOR = _calculateDomainSeparator(block.chainid);


/// ... Return the DOMAIN_SEPARATOR.
return block.chainid == deploymentChainId ? _DOMAIN_SEPARATOR : _calculateDomainSeparator(block.chainid);


/// ...   IERC2612-permit abi.encode
block.chainid == deploymentChainId ? _DOMAIN_SEPARATOR : _calculateDomainSeparator(block.chainid),
~~~

**Comment by roninjin10** [comment]
Ethers.js and web3.js do not support this EIP. That is a strong reason to not adopt it in wagmi especially considering it comes with negative tradeoffs in the form of potential casing bugs/confusion. Will leave this open for now though but maybe we close this issue.
